bc_obps/rls/utils/manager.py

```diff
@@ -0,0 +1,68 @@
+from django.conf import settings
+from django.db import connection
+from django.db.backends.utils import CursorWrapper
+from psycopg.sql import Identifier, SQL
+from rls.enums import RlsRoles
+from django.apps import apps
+from rls.utils.m2m import M2mRls
+
+
+class RlsManager:
+    @staticmethod
+    def revoke_all_privileges() -> None:
+        # Convert enum values into an SQL-safe list
+        role_identifiers = [Identifier(role.value) for role in RlsRoles]
+        with connection.cursor() as cursor:
+            drop_owned_by_query = SQL('drop owned by {}').format(SQL(', ').join(role_identifiers))
+            grant_usage_query = SQL("grant usage on schema erc to {}").format(SQL(', ').join(role_identifiers))
+            cursor.execute(drop_owned_by_query)
+            cursor.execute(grant_usage_query)
+
+            # Grant usage and all privileges on all tables in schema erc_history to public(We don't care about this schema)
+            cursor.execute("grant usage on schema erc_history to public")
+            cursor.execute("grant all privileges on all tables in schema erc_history to public")
+            # Grant usage and select privileges on all tables in schema common to public
+            cursor.execute("grant usage on schema common to public")
+            cursor.execute("grant select on all tables in schema common to public")
+
+    @classmethod
+    def apply_rls(cls) -> None:
+        apps_to_apply_rls = [app for app in settings.LOCAL_APPS if app != 'rls']
+        for app_name in apps_to_apply_rls:
+            for model_name in apps.all_models[app_name]:
+                cls.apply_rls_for_model(app_name, model_name)
+
+    @classmethod
+    def apply_rls_for_model(cls, app_name: str, model_name: str) -> None:
+        model = apps.all_models[app_name][model_name]
+        if hasattr(model, 'Rls'):
```

Comment:
I wonder if this is something we'd want to remove - we would want to enforce all our models to have an RLS setup, and models missing that attribute (or inner class or whatever we choose) should throw an error

Reply:
We’ve got that covered in two other ways:
1. Once we remove `if settings.RLS_FLAG is False:` in the `class Rls(ModelBase):`, it will raise a `NotImplementedError` if a model inheriting from `BaseModel` doesn’t implement RLS.
2. The `test_all_models_have_rls` test checks all models and their M2M fields.